android-core/src/main/java/com/mparticle/internal/AppStateManager.java

```diff
@@ -368,10 +372,19 @@ private void logBackgrounded() {
             instance.Internal().getKitManager().onApplicationBackground();
             mCurrentActivityName = null;
             Logger.debug("App backgrounded.");
+            scheduleBackgroundJob();
             mInterruptionCount.incrementAndGet();
         }
     }
 
+    private void scheduleBackgroundJob() {
```

Comment:
What is the purpose of this separate task vs just using the repeating one? Weren't we going to merge them into one or am I misremembering the details?

Reply:
This function is scheduling the job.
In terms why ONE_SHOT vs PERIODIC, the idea is to start a ONE_SHOT when going on background to prevent recurrently uploading stuff with no need. When the ONE_SHOT triggers we will set a recurrent one based on the uploadInterval, being active and having stuff to send. Its a combination of both.

Reply:
Sounds good, thanks for the explanation.